osu.Game/Skinning/LegacySkin.cs

```diff
@@ -81,15 +89,31 @@ public override Drawable GetDrawableComponent(string componentName)
                         };
             }
 
-            // temporary allowance is given for skins the fact that stable handles non-animatable items such as hitcircles (incorrectly)
-            // by (incorrectly) displaying the first frame of animation rather than the non-animated version.
-            // users have used this to "hide" certain elements like hit300.
-            var texture = GetTexture($"{componentName}-0") ?? GetTexture(componentName);
+            var texture = GetTexture($"{componentName}-0");
 
-            if (texture == null)
-                return null;
+            if (texture != null && animatable)
+            {
+                var animation = new TextureAnimation { DefaultFrameLength = animationData.frametime };
+
+                for (int i = 1; texture != null; i++)
+                {
+                    animation.AddFrame(texture);
+                    texture = GetTexture($"{componentName}-{i}");
+                }
+
+                animation.Repeat = animationData.looping;
 
-            return new Sprite { Texture = texture };
+                return animation;
+            }
+            else
+            {
+                texture = GetTexture(componentName);
+
+                if (texture == null)
+                    return null;
+
+                return new Sprite { Texture = texture };
```

Comment:
```suggestion
                return texture != null ? new Sprite { Texture = GetTexture(componentName) } : null;
```

Reply:
this is important. without returning actual `null` the skin fallback logic will not work

Reply:
But I am returning ``null`` a few lines above this. His suggestion is pure syntax sugar...